truss/server/inference_server.py

```diff
@@ -16,8 +17,7 @@ class ConfiguredTrussServer:
 
     def __init__(self, config_path: str, port: int):
         self._port = port
-        with open(config_path, encoding="utf-8") as config_file:
-            self._config = yaml.safe_load(config_file)
+        self._config = TrussConfig.from_yaml(Path(config_path)).to_dict(verbose=True)
```

Comment:
this is a pretty big change, we'll have to change the types on the TrussServer to accept a `Config` object  & make sure everything that needs config accesses in this way now

Reply:
it's probably not super clear because its all on one line but I'm calling `to_dict` at the end so that none of the types change.

Reply:
I essentially moved the parsing and spitting out the config from the time we build the context for the docker build to server startup

Reply:
ah oops yeah I missed the `to_dict`. It would actually be cool if we could pass this object around instead of the dict, but we can do that separately.

Reply:
Let's leave a TODO note?

Reply:
I'll make a linear ticket